packages/circuit-ui/components/Checkbox/Checkbox.stories.tsx

```diff
@@ -68,9 +66,26 @@ export const Disabled = (args: CheckboxProps) => (
 
 Disabled.args = {
   name: 'disabled',
-  value: 'true',
+  value: 'any value',
   disabled: true,
 };
+export const Indeterminate = (args: CheckboxProps) => (
+  <CheckboxWithState
+    {...args}
+    ref={(el: HTMLInputElement) => {
+      if (el) {
+        // eslint-disable-next-line no-param-reassign
+        el.indeterminate = true;
+      }
+    }}
```

Comment:
This PR was just about making the style of the Checkbox as useful as the browser checkbox,

I don't know how about the way we would like to use it.
So I just used the native behavior, allowing the `:indeterminate` state to show up.

Reply:
Got it 👍 

This is how they do it over at Spectrum:

https://github.com/adobe/react-spectrum/blob/b64e64f28b3d06a69ad2c5ae0dd994c3a07d4099/packages/%40react-aria/checkbox/src/useCheckbox.ts#L42-L49

I kind of like this because it abstracts the complexity of "yeah a checkbox can be indeterminate but you _have to_ use JS for it, and in React that means you need to use the ref to the underlying HTML checkbox node" into the design system, and then developers could just set a boolean prop `isIndeterminate` or `indeterminate` to get the styles.

Happy to merge this though (since as you said, this was always meant for styles) and then make a follow-up PR to add the convenience prop. Let me know what you prefer (in this PR or as follow-up) 🙂

Reply:
It is, here too, stated that indeterminism is presentational only.
https://github.com/adobe/react-spectrum/blob/b64e64f28b3d06a69ad2c5ae0dd994c3a07d4099/packages/%40react-types/checkbox/src/index.d.ts#L67
As we thought, it is the responsability of the developper to handle the checked state according to its business logic.

Therefore, I'm not sure the property should be available like the checked one...